src/backend/joanie/tests/payment/test_backend_lyra.py

```diff
@@ -2075,3 +2076,209 @@ def test_payment_backend_lyra_cancel_or_refund_with_wrong_transaction_reference_
+        order = OrderFactory(
+            state=ORDER_STATE_PENDING,
+            owner=owner,
+            main_invoice=InvoiceFactory(),
+            payment_schedule=[
+                {
+                    "id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
+                    "amount": "200.00",
+                    "due_date": "2024-01-17",
+                    "state": PAYMENT_STATE_PAID,
+                },
+                {
+                    "id": "fa17d7b8-3b86-4755-ac78-bc039018d696",
+                    "amount": "300.00",
+                    "due_date": "2024-02-17",
+                    "state": PAYMENT_STATE_PENDING,
+                },
+            ],
+        )
+
+        with self.open("lyra/responses/is_already_paid.json") as file:
+            json_response = json.loads(file.read())
+
+        json_response["answer"]["transactions"][0]["status"] = "UNPAID"
+
+        responses.add(
+            responses.POST,
+            "https://api.lyra.com/api-payment/V4/Order/Get",
+            headers={
+                "Content-Type": "application/json",
+            },
+            match=[
+                responses.matchers.header_matcher(
+                    {
+                        "content-type": "application/json",
+                        "authorization": "Basic Njk4NzYzNTc6dGVzdHBhc3N3b3JkX0RFTU9QUklWQVRFS0VZMjNHNDQ3NXpYWlEyVUE1eDdN",
+                    }
+                ),
+                responses.matchers.json_params_matcher({"orderId": str(order.id)}),
+            ],
+            status=200,
+            json=json_response,
+        )
+
+        self.assertFalse(backend.is_already_paid(order, order.payment_schedule[1]))
+
+        # installment is marked as refused
+        order.refresh_from_db()
+        self.assertEqual(order.payment_schedule[1]["state"], PAYMENT_STATE_REFUSED)
+
+    @responses.activate(assert_all_requests_are_fired=True)
+    def test_payment_backend_lyra_is_already_paid_error(self):
+        """
+        When backend checks if an installment has already been paid, it should return False
+        if the payment has not been made and should update the installment state to refused.
+        """
+        backend = LyraBackend(self.configuration)
+        owner = UserFactory(email="john.doe@acme.org")
+        UserAddressFactory(owner=owner)
+        order = OrderFactory(
+            state=ORDER_STATE_PENDING,
+            owner=owner,
+            main_invoice=InvoiceFactory(),
+            payment_schedule=[
+                {
+                    "id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
+                    "amount": "200.00",
+                    "due_date": "2024-01-17",
+                    "state": PAYMENT_STATE_PAID,
+                },
+                {
+                    "id": "fa17d7b8-3b86-4755-ac78-bc039018d696",
+                    "amount": "300.00",
+                    "due_date": "2024-02-17",
+                    "state": PAYMENT_STATE_PENDING,
+                },
```

Comment:
```suggestion
                {
                    "id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
                    "amount": "200.00",
                    "due_date": "2024-01-17",
                    "state": PAYMENT_STATE_PENDING,
                },
                {
                    "id": "fa17d7b8-3b86-4755-ac78-bc039018d696",
                    "amount": "300.00",
                    "due_date": "2024-02-17",
                    "state": PAYMENT_STATE_PENDING,
                },
```
I think this case should only happen when the debit happens on the very first installment of the order. In that case, we can not have the information of the order reference at the payment provider side.

If it concerns the second installment, then, by calling Order/Get, we might be able to find the first paid installment information because they both have the same order id.